DependencyInjection/Compiler/IdGeneratorPass.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\DoctrineBundle\Mapping\ClassMetadataFactory;
+use Doctrine\Bundle\DoctrineBundle\Mapping\MappingDriver;
+use Doctrine\ORM\Mapping\ClassMetadataFactory as ORMClassMetadataFactory;
+use Symfony\Component\DependencyInjection\Alias;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Compiler\ServiceLocatorTagPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class IdGeneratorPass implements CompilerPassInterface
+{
+    const ID_GENERATOR_TAG  = 'doctrine.id_generator';
+    const CONFIGURATION_TAG = 'doctrine.orm.configuration';
+
+    public function process(ContainerBuilder $container): void
+    {
+        $generatorIds     = array_keys($container->findTaggedServiceIds(self::ID_GENERATOR_TAG));
+        $configurationIds = array_keys($container->findTaggedServiceIds(self::CONFIGURATION_TAG));
+
+        foreach ($configurationIds as $id) {
+            $container->getDefinition($id)->clearTag(self::CONFIGURATION_TAG);
+        }
+
+        // when ORM is not enabled
+        if (! $container->hasDefinition('doctrine.orm.configuration') || ! $generatorIds) {
+            return;
+        }
+
+        $generatorRefs = array_map(static function ($id) {
+            return new Reference($id);
+        }, $generatorIds);
+
+        $ref = ServiceLocatorTagPass::register($container, array_combine($generatorIds, $generatorRefs));
+        $container->setAlias('doctrine.id_generator_locator', new Alias((string) $ref, false));
+
+        foreach ($configurationIds as $id) {
+            $methodCalls        = $container->getDefinition($id)->getMethodCalls();
+            $metadataDriverImpl = null;
+
+            foreach ($methodCalls as $i => [$method, $arguments]) {
+                if ($method === 'setMetadataDriverImpl') {
+                    $metadataDriverImpl = (string) $arguments[0];
+                }
+
+                if ($method !== 'setClassMetadataFactoryName') {
+                    continue;
+                }
+
+                if ($arguments[0] !== ORMClassMetadataFactory::class && $arguments[0] !== ClassMetadataFactory::class) {
```

Comment:
What's the purpose of this block? Body looks fishy to me and it's not covered by tests.

Reply:
The purpose of this block is to *not* wire the `MappingDriver` decorators if the `ClassMetadataFactory` doesn't know how to handle the `instance` key in the `$customGeneratorDefinition`. But I'm fine removing the check if you'd prefer.